tests/e2e/cucumber/features/integrations/reshare.ocis.feature

```diff
@@ -0,0 +1,54 @@
+Feature: reshare
+
+  Background:
+    Given "Admin" disables share auto accepting
+
+  Scenario: re-sharing
+    Given "Admin" creates following users
+      | id    |
+      | Alice |
+      | Brian |
+      | Carol |
+    And "Admin" creates following group
+      | id      |
+      | sales   |
+      | finance |
+    And "Admin" adds user to the group
+      | user  | group |
+      | Carol | sales |
+    
+    And "Alice" logs in
+    And "Alice" opens the "files" app
+    And "Alice" navigates to the personal space page
+    And "Alice" creates the following resources
+      | resource         | type   |
+      | folder_to_shared | folder |
+    When "Alice" shares the following resource using the sidebar panel
+      | resource         | user  | role   |
+      | folder_to_shared | Brian | editor |
+    And "Alice" logs out
+    
+    And "Brian" logs in
+    And "Brian" opens the "files" app
+    And "Brian" navigates to the shared with me page
+    And "Brian" accepts the following share
+      | name             |
+      | folder_to_shared |
+    And "Brian" reshares the following resource
+      | resource         | user  | role   |
+      | folder_to_shared | Carol | viewer |
+    And "Brian" logs out
+    
+    And "Carol" logs in
+    And "Carol" opens the "files" app
+    And "Carol" navigates to the shared with me page
+    And "Carol" accepts the following share
+      | name             |
+      | folder_to_shared |
+    And "Carol" reshares the following resource
+      | resource         | user  | role   |
+      | folder_to_shared | Alice | viewer |
```

Comment:
I didn't know it was possible to reshare the resource to the person to who the resource belongs with a different roles. Is this expected?

I tried to play around with the UI 
1. Alice accepts the share request
2. Alice as the owner of the resource creates a sub folder inside `folder_to_shared` like `test`
3. The sub-folder is displayed in shared with me inside `folder_to_shared`
4. If Alice goes inside folder_to_shared->tests from shared with me (this folder is shared with role viewer), she can create resources inside it but not directly inside `folder_to_shared`

Probably not related to this PR. Seems like a bug.

Reply:
I didn't try to check this case. It's too complicated. It doesn't seem realistic.
I checked only that user can re-share re-shared resource. I did not create the 4th user

Reply:
I'll create a separate issue for this case

Reply:
issue created here https://github.com/owncloud/ocis/issues/4336